### PR TITLE
Set all setter definitions from source package

### PR DESCRIPTION
**Pull request by phanimarupaka**
@pwittrock 

This PR is to SetAllSetterDefinitions in source folder Kptfile to the destination folder files.

### CODE REVIEW

File: kyaml/setters2/settersutil/fieldsetter.go [hunk 1]
```diff
@@ -68,3 +68,36 @@ func (fs FieldSetter) Set(openAPIPath, resourcesPath string) (int, error) {
 	}.Execute()
 	return s.Count, err
 }
+
+// setAllSetterDefinitions reads all the Setter Definitions from Kptfile in source
```
Comment (pwittrock) [inline kyaml/setters2/settersutil/fieldsetter.go:72]: nit: SetAll...

Comment (phanimarupaka) [reply kyaml/setters2/settersutil/fieldsetter.go:72]: Done

Comment (pwittrock) [reply kyaml/setters2/settersutil/fieldsetter.go:72]: Capitalization?

File: kyaml/setters2/settersutil/fieldsetter.go [hunk 2]
```diff
@@ -68,3 +68,36 @@ func (fs FieldSetter) Set(openAPIPath, resourcesPath string) (int, error) {
 	}.Execute()
 	return s.Count, err
 }
+
+// SetAllSetterDefinitions reads all the Setter Definitions from OpenAPI in source
+// package folder and sets all setter values in destination package folder
+func SetAllSetterDefinitions(args []string) error {
```
Comment (pwittrock) [inline kyaml/setters2/settersutil/fieldsetter.go:74]: How about:

```go
func SetAllSetterDefinitions(openAPI string, dir ...string) error 
```

File: kyaml/setters2/settersutil/fieldsetter.go [hunk 3]
```diff
@@ -68,3 +68,36 @@ func (fs FieldSetter) Set(openAPIPath, resourcesPath string) (int, error) {
...
+	// get all the setter definitions from package
+	l := setters2.List{}
+	err := l.List(sourcePkgOpenAPIPath, sourcePkgPath)
+	if err != nil {
+		return err
+	}
+
+	// for each setter definition set the setter values in original and updated packages
+	//TODO(pmarupaka): optimize to perform all the setters in single pass instead of N passes
+	for _, sd := range l.Setters {
```
Comment (pwittrock) [inline kyaml/setters2/settersutil/fieldsetter.go:89]:
```go
for _, sd := range l.Setters {
  for _, dir := range dirs {
```

File: kyaml/setters2/settersutil/fieldsetter.go [hunk 4]
```diff
@@ -68,3 +68,36 @@ func (fs FieldSetter) Set(openAPIPath, resourcesPath string) (int, error) {
...
+		fs := FieldSetter{
+			Name:        sd.Name,
+			Value:       sd.Value,
+			ListValues:  sd.ListValues,
+			Description: sd.Description,
+			SetBy:       sd.SetBy,
+		}
+		_, err = fs.Set(destPkgOpenAPIPath, destPkgPath)
```
Comment (pwittrock) [inline kyaml/setters2/settersutil/fieldsetter.go:97]:
```go
fs.Set(openAPI, dir)
```

File: kyaml/setters2/settersutil/fieldsetter.go (hunk 3 above)
Comment (phanimarupaka) [reply kyaml/setters2/settersutil/fieldsetter.go:85]: Done.

File: kyaml/setters2/settersutil/fieldsetter.go (hunk 2 above)
Comment (phanimarupaka) [reply kyaml/setters2/settersutil/fieldsetter.go]: Done

**Comment by pwittrock** [comment]
Need to follow up with a few things, but will take this as it is for now.

**Pull request merged**